payments-core/src/main/java/com/stripe/android/networking/StripeApiRepository.kt

```diff
@@ -1246,6 +1254,36 @@ internal class StripeApiRepository @JvmOverloads internal constructor(
             .plus(createExpandParam(expandFields))
     }
 
+    private fun buildPaymentUserAgentPair(attribution: Set<String> = emptySet()) =
+        PAYMENT_USER_AGENT to
+            setOf("stripe-android/${Stripe.VERSION_NAME}")
+                .plus(productUsageTokens)
+                .plus(attribution)
+                .joinToString(";")
+
+    /**
+     *  Add payment_user_agent to the map if it contains Payment Method data,
+     *  including attribution from [paymentMethodCreateParams] or [sourceParams].
+     */
+    private fun maybeAddPaymentUserAgent(
```

Comment:
should there be any tests around this logic?

Reply:
It's being tested through the test cases I added to `StripeApiRepositoryTest`, I think that's better because those are the actual public methods.